doc/design/networking/Investigation-on-NSX-integration.md

```diff
@@ -0,0 +1,109 @@
+# Investigation on NSX micro-segmentation via Docker network create
+This document is for this issue: https://github.com/vmware/vic/issues/3936
+
+
+## A Working Installation of NSX-v
+I documented how to install and configure NSX with Vsphere: How to set up VM-to-VM communication on NSX logical switch with Nimbus
+https://confluence.eng.vmware.com/pages/viewpage.action?pageId=209964551
+
+## An Overview of Workflow
+
+### Prerequisite
+VSphere admin already has NSX installed and configured on VCenter. That means there is an NSX logical switch created by the admin, which automatically creates a port group on distributed switch.
+So this port group can be used as a bridge network for `vic-machine create`
+
+![Distributed Switch with logic switch added] (pics/dswitch.png)
+
+### Create VCH 
+specifying 'bridge-network' as the port group 'vxw-dvs-55-virtualwire-1-sid-5000-logical-switch-1':
+  * I use this VCH to create containers, they are automatically added to this port group (i.e. logical switch)
+  * containerVMs get their IPs in the internal network
```

Comment:
This would depends on the NSX configuration and topology. NSX edge can act like a DHCP server, but can only serve interfaces directly connected to it. or NSX Distrubted Logical Router(DLR) can have external DHCP servers via DHCP relay configuration. However, we cannnot configure a DHCP on an ESG and then use the DHCP Relay function from the DLR to get an IP address from the ESG.